### PR TITLE
Tax Number Formatter

### PR DESCRIPTION
**Pull request by olsza**

### CODE REVIEW

File: src/Collection/TaxNumberFormatter.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                strlen($country)
+            )
+        );
+
+        return $prefix === $country
+            ? $country . substr(
+                $tax_number,
+                strlen($country)
+            )
+            : $country . $tax_number;
```
Comment (michael-rubel) [inline src/Collection/TaxNumberFormatter.php]: I would extract bigger blocks of code to methods and probably make use of Laravel's `Str` class to operate on strings.

Comment (olsza) [reply src/Collection/TaxNumberFormatter.php]: ok

File: src/Collection/TaxNumberFormatter.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    /**
+     * @param Collection $items
+     * @return string
+     */
+    private function getCountry(Collection $items): string
+    {
+        return (string) Str::of($items->get($this->key_country))
+            ->upper();
```
Comment (michael-rubel) [inline src/Collection/TaxNumberFormatter.php]:
```suggestion
        return (string) Str::of(
            $items->get($this->key_country)
        )->upper();
```

File: src/Collection/TaxNumberFormatter.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        $tax_number = $this->cleanupTaxNumber($items);
+
+        if (empty($items->get($this->country_key))) {
+            return $tax_number;
+        }
+
+        return $this->getFullTaxNumber(
+            $tax_number,
+            $this->getCountry($items)
+        );
```
Comment (michael-rubel) [inline src/Collection/TaxNumberFormatter.php]:
```suggestion
        return empty($items->get($this->country_key))
            ? $tax_number
            : $this->getFullTaxNumber(
                $tax_number,
                $this->getCountry($items)
            );
```

**Review by michael-rubel — APPROVED** [review]

**Pull request merged**